test/util.go

```diff
@@ -115,6 +117,11 @@ func WaitForMeshWithMessage(t *testing.T, topic string, hosts ...host.Host) []*p
 		select {
 		case <-done:
 		case <-time.After(waitForMeshTimeout):
+			if hackyRecursionLimit > 0 {
+				hackyRecursionLimit -= 1
+				return WaitForMeshWithMessage(t, topic, hosts...)
+			}
```

Comment:
Is there any reason to expect this to work over a longer wait time? Has that ever been observed?

Reply:
no, I think if it doesn't complete in <1s then it will fail. Not sure why yet. Only happens in CI from what I can tell :(